src/coreclr/System.Private.CoreLib/src/System/Collections/Generic/ComparerHelpers.cs

```diff
@@ -122,41 +85,13 @@ internal static object CreateDefaultEqualityComparer(Type type)
             else if (type.IsEnum)
             {
                 // The equality comparer for enums is specialized to avoid boxing.
-                result = TryCreateEnumEqualityComparer(runtimeType);
-            }
-
-            return result ?? CreateInstanceForAnotherGenericParameter((RuntimeType)typeof(ObjectEqualityComparer<object>), runtimeType);
-        }
-
-        /// <summary>
-        /// Creates the default <see cref="EqualityComparer{T}"/> for an enum type.
-        /// </summary>
-        /// <param name="enumType">The enum type to create the default equality comparer for.</param>
-        private static object? TryCreateEnumEqualityComparer(RuntimeType enumType)
-        {
-            Debug.Assert(enumType != null);
-            Debug.Assert(enumType.IsEnum);
-
-            // See the METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST and METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST_LONG cases in getILIntrinsicImplementation
-            // for how we cast the enum types to integral values in the comparer without boxing.
 
-            TypeCode underlyingTypeCode = Type.GetTypeCode(Enum.GetUnderlyingType(enumType));
-
-            // Depending on the enum type, we need to special case the comparers so that we avoid boxing.
-            switch (underlyingTypeCode)
```

Comment:
This is going to change the equals behavior for enums backed by floats, etc. It is probably ok. Do we have a test for it that will need to change?

Reply:
I'm not sure, I assumed it to be fine since NativeAOT VM already handled those like this.

Reply:
NativeAOT does not support the non-standard enums.